client/post-editor/editor-preview/index.jsx

```diff
@@ -47,6 +48,13 @@ const EditorPreview = React.createClass( {
 		) ) {
 			this.setState( { iframeUrl: this.getIframePreviewUrl() } );
 		}
+
+		if (
+			this.state.iframeUrl !== 'about:blank' &&
+			prevProps.showPreview !== this.props.showPreview
```

Comment:
Extract this check to a method such as `didShowOrHidePreview`. Then, you could move it to the conditional above, with the other `did` checks. That way, we only have only call to `setState`, and the code is a bit more readable.

Reply:
This is not logically equivalent. What we really want is to set the preview URL's hash only if it is not currently set to `about:blank`. What you propose would cause the URL to be changed from `about:blank` to the actual preview URL any time the preview is hidden or shown, rather than waiting for any pending save to complete.

Reply:
I thought we would also include the `about:blank` check in the `didShowOrHidePreview`. Maybe I'm still missing something, but that only highlights that this method is getting difficult to quickly understand. :-)

Reply:
Ah, yes, I agree that should work fine.

It would be nice to do some more extensive refactoring here:  ideally the preview URL should just be a function of the current props, which eliminates a lot of this complexity.  However, I think that is a riskier change and best saved for a separate task.

Reply:
Thanks for the suggestion and for clarifying what you meant.  I agree the code looks better this way :+1: